src/rgw/rgw_rest_s3.cc

```diff
@@ -3579,13 +3579,23 @@ RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
 
    // Non-website mode    // Non-website mode
   if (get_data) {   
-    if (list_type == 1) {
-       return new RGWListBucket_ObjStore_S3;     
-    } else if(list_type == 2) {
-      return new RGWListBucket_ObjStore_S3v2;
-    } } else {
-    return new RGWStatBucket_ObjStore_S3;    
-  }   }
+    switch(list_type) {
+        case 1:
+	  return new RGWListBucket_ObjStore_S3;
+	  break;
+        case 2:
+	  return new RGWListBucket_ObjStore_S3v2;
+	  break;
+        default:
+	  ldpp_dout(s, 5) << __func__ << ": unsupported list-type " << list_type << dendl;
+	  return new RGWListBucket_ObjStore_S3;
+	  break;
+    }
+  }
+  else {
```

Comment:
i am curious, if `get_data` is false, why bother getting "list-type" and dropping it on the floor? can we take this opportunity to add an other commit to address this issue in this PR?

Reply:
please move the `else` up into previous line.